### ISSUE TITLE
自宅鯖へのデプロイフロー構築

### ISSUE THREAD

**Issue by supermomonga** [supermomonga/kokoro-io#160]
- [ ] docker-compose.production.yml による手動デプロイ
- [ ] デプロイの自動化（どうやって？）

**Comment by kamichidu** [comment]
> デプロイの自動化
手軽にやるなら、
1. 自宅鯖内でcronなりでdocker-compose up --no-deps -dを定期実行させる
2. masterにpushやtagが切られたタイミングで、docker hubにビルドしたイメージをアップ

ちゃんとやるなら、自宅鯖のdockerdを外部に公開して、CIサービス内から自宅鯖のdockerdに更新命令を出す

とか

**Comment by supermomonga** [comment]
割とリアルタイム性が欲しい気がするので、

1. GitHub の release ブランチにプッシュされたら CI をキック
2. CI が鯖に ssh して `git pull && docker-compose build && docker-compose up -d`

こういうのを考えてた

**Comment by kamichidu** [comment]
dockerdはremote hostのdockerから命令出せるので、そっち使うほうが筋が良さそう。

**Comment by supermomonga** [comment]
オッ、なるほど。

**Comment by kamichidu** [comment]
それとdocker-composeを使うと、ホストが再起動したりしたときの自動起動あたりで詰まってしまうので、1ホストだけだけどdocker swarm使ったほうが良い気がします。
docker swarmはcomposefile食えるので。

**Comment by kamichidu** [comment]
ただdocker swarmはdocker registryの存在を暗黙に仮定しているので、まず手軽にやるならdocker-composeでbuildからupまでやるのが最初は良いかなと思います。特にまだ開発中ですし。

**Comment by supermomonga** [comment]
そうですね、何となく考えてる予定としては、Closed Beta 中は単純な docker-compose, ダウンタイムあり, データの消失あり という感じで、Open Beta 時には swarm か kubernetes か Rancher か、そのあたりのを導入しようかと思ってます

**Comment by kamichidu** [comment]
個人的に、docker使うなら、swarmは十分本番投入できるレベルだと思うんで、swarm推しです。
Rancherは使ったことないですけど、悪くない選択肢なのかなぁ。
